torch/_functorch/aot_autograd.py

```diff
@@ -1626,10 +1975,22 @@ def aot_dispatch_autograd(flat_fn, flat_args: List[Any], aot_config: AOTConfig):
         flat_args, _fw_metadata.input_info
     )
 
-    joint_forward_backward = create_joint_forward_backward_functionalized(
+    # pre-compute, so we can bail out quickly in the hotpath
+    metadata_ = CompiledRuntimeMetadata(
```

Comment:
I think that metadata organization could use a cleanup PR - currently planning on revisiting that later